recipes-nodes/config-watchdog/config-watchdog.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/sh
+### BEGIN INIT INFO
+# Provides:          config-watchdog
+# Required-Start:    $network
+# Required-Stop:
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: config watchdog
+# Description:       Pulls the config from BuilderHub and reloads repsective services
+### END INIT INFO
+
+set -e
+
+SYSTEM_API_FIFO="/var/volatile/system-api.fifo"
+RBUILDER_PERSISTENT_DIR="/persistent/rbuilder"
+RBUILDER_STATUS_FLAG_FILE="${RBUILDER_PERSISTENT_DIR}/rbuilder.enabled"
+# locally exposed BuilderHub API proxied through CVM proxy
+BHUB_URL_PROXIED="http://localhost:7937/api/l1-builder/v1/configuration"
+BHUB_CONFIG_FILE="/etc/init-config-watchdog.json"
+
+function cleanup() {
+  rm -f "${BHUB_CONFIG_FILE}"
+}
+
+trap cleanup EXIT
+
+log() {
+    if [ -p $SYSTEM_API_FIFO ]; then
+        date_log() {
+            echo -n ""
+        }
+    else
+        date_log() {
+            echo -n "$(date --iso-8601=seconds): "
+        }
+    fi
+    echo "$(date_log)$1" | tee -a $SYSTEM_API_FIFO
+}
+
+function toggle_rbuilder() {
+  local bhub_config_file="$1"
+  local rbuilder_enabled=$(cat "${bhub_config_file}" | jq -r '.rbuilder.enabled')
+  if [ ! -d "${RBUILDER_PERSISTENT_DIR}" ]; then
+    umask 0022
+    mkdir -p $RBUILDER_PERSISTENT_DIR
+  fi
+
+  umask 0133
+  if [ "$rbuilder_enabled" = "true" ]; then
+    if [ ! -f "${RBUILDER_STATUS_FLAG_FILE}" ]; then
+      touch "${RBUILDER_STATUS_FLAG_FILE}"
+      log "rbuilder-status-watchdog: rbuilder is enabled in BuilderHub"
+      service rbuilder restart
+    fi
+  else
+    if [ -f "${RBUILDER_STATUS_FLAG_FILE}" ]; then
+      rm -f "${RBUILDER_STATUS_FLAG_FILE}"
+      log "rbuilder-status-watchdog: rbuilder is disabled in BuilderHub"
+      service rbuilder stop
+    fi
+  fi
+}
+
+function update_subsidies_config() {
+  local bhub_config_file="$1"
+  /usr/bin/render-config.sh --unsafe "${bhub_config_file}" /etc/rbuilder-bidding/bidding-service.toml.mustache > /tmp/bidding-service.toml.tmp
+
+  if ! diff -q /tmp/bidding-service.toml.tmp /etc/rbuilder-bidding/bidding-service.toml > /dev/null; then
+    mv /tmp/bidding-service.toml.tmp /etc/rbuilder-bidding/bidding-service.toml
+    podman kill --signal=SIGHUP rbuilder-bidding
+    log "update-subsisidies-config: Updated bidding service config"
+  fi
+}
+
+case "$1" in
+  start)
+    (umask 0177 && touch "${BHUB_CONFIG_FILE}")
```

Comment:
why is this step needed? a comment would be nice

Reply:
To create a file with permissions equal to `0600` making it RW for root only.